akka-actor-typed/src/main/scala/akka/actor/typed/internal/InterceptorImpl.scala

```diff
@@ -74,9 +76,10 @@ private[akka] final class InterceptorImpl[O, I](
     new InterceptorImpl(interceptor, newNested)
 
   override def receive(ctx: typed.TypedActorContext[O], msg: O): Behavior[O] = {
-    val interceptMessageType = interceptor.interceptMessageType
+    // TODO performance optimization could maybe to avoid isAssignableFrom if interceptMessageClass is Class[Object]?
+    val interceptMessageClass = interceptor.interceptMessageClass
     val result =
-      if (interceptMessageType == null || interceptMessageType.isAssignableFrom(msg.getClass))
+      if ((interceptMessageClass ne null) && interceptor.interceptMessageClass.isAssignableFrom(msg.getClass))
```

Comment:
This has changed from before, now `null` does _not_ intercept, while before it meant intercept all messages, was that intentional?

Reply:
yes, intentional. The previous `null`, which was also the default was unsafe as we have seen.

The new `null` is needed for when the interceptor shouldn't intercept messages but only signals. That is provided as end user api via `BehaviorSignalInterceptor`